TreasureDataTests/IntegrationTests.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  TreasureDataSwiftTests.swift
+//  TreasureDataTests
+//
+//  Created by Huy TD on 1/27/19.
+//  Copyright © 2019 Huy TD. All rights reserved.
+//
+
+import XCTest
+
+class IntegrationTests: XCTestCase {
+
+    var sdkClient: TreasureData!
+    static var api: TDAPI!
+    static var apiKey: String!
+
+    static let TargetDatabase = "ios_it"
+    static var sessionPrefix = uuid()
+
+    /** Tables used in a single test */
+    var tempTables: [String] = []
+
+    override class func setUp() {
+        apiKey = ProcessInfo.processInfo.environment["TD_DEV_MASTER_KEY"]
+        api = TDAPI.init(endpoint: "https://api-development.treasuredata.com", apiKey: apiKey)
+        if (!(try! api.isDatabaseExist(TargetDatabase))) {
+            fatalError("Either the apiKey is invalid or the target database \(TargetDatabase) is not exist!")
+        }
+    }
+
+    override class func tearDown() {
+        IntegrationTests.api = nil
+    }
+
+    override func setUp() {
+        TreasureData.initializeApiEndpoint("https://in-development.treasuredata.com")
+        sdkClient = TreasureData(apiKey: IntegrationTests.apiKey)
+        sdkClient.defaultDatabase = IntegrationTests.TargetDatabase
+        tempTables = []
+    }
+
+    override func tearDown() {
+        let remoteTables = Set<String>(IntegrationTests.api.listTables(database: IntegrationTests.TargetDatabase))
+        for sessionTable in tempTables {
+            if remoteTables.contains(sessionTable) {
+                try! IntegrationTests.api.deleteTable(database: IntegrationTests.TargetDatabase, table: sessionTable)
+            } else {
+                print("WARN: Skip deleting table '\(IntegrationTests.TargetDatabase).\(sessionTable)' as it is not exist!")
+            }
+        }
+        sdkClient.disableAppLifecycleEvent()
+        sdkClient = nil
+    }
+
+    func testCustomEvent() {
+        sdkClient.enableCustomEvent()
+        let table = newTempTable()
+        let event = sdkClient.addEvent(["message": "Guten Morgen!"], table: table)
+        sdkClient.uploadEvents()
+        if event != nil {
+            let result = try! IntegrationTests.api.stubbornQuery("select message from \(table) limit 1", database: IntegrationTests.TargetDatabase)
```

Comment:
How long to loop before receiving the result ?

Reply:
I observed it could span from somewhere from 100 to 300 secs, usually it around ~200s.